src/qt/sendcoinsdialog.cpp

```diff
@@ -791,8 +791,13 @@ void SendCoinsDialog::useAvailableBalance(SendCoinsEntry* entry)
     // Include watch-only for wallets without private key
     m_coin_control->fAllowWatchOnly = model->wallet().privateKeysDisabled() && !model->wallet().hasExternalSigner();
 
+    // Same behavior as send: if we have selected coins, only obtain their available balance.
+    // future, introduce a checkbox to customize this value.
+    CCoinControl coin_control = *m_coin_control;
+    coin_control.m_allow_other_inputs = !coin_control.HasSelected();
```

Comment:
The `m_allow_other_inputs` flag is different to the `fAllowWatchOnly` flag.

`fAllowWatchOnly` set is actually redundant here as `privateKeysDisabled()` and `hasExternalSigner()` are wallet constants. It can be set only once, when the wallet is set in the GUI and don't be re-set again.

And, at the contraire, the `m_allow_other_inputs` flag can change as it depends on whether the user manually selected coins or not, which could be modified at any time. Thus why I would try to not set the coin control member here.

The best path would be to set `m_allow_other_inputs` at the same moment when the user manually selects/deselect coins in the coin control dialog. But.. I think that it's better to do that in a follow-up to not expand this PR much.

Reply:
Ah, maybe add a comment:

```cpp
// Copy to avoid modifying m_coin_control 
```

Reply:
Sure. Doesn't hurt to add a comment.

Reply:
pushed